platform/linux/mod.rs

```diff
@@ -278,16 +286,10 @@ impl UnixSender {
 
                 if result <= 0 {
                     let error = UnixError::last();
-                    if error.0 == libc::ENOBUFS && bytes_to_send > 2000 {
+                    if error.0 == libc::ENOBUFS
+                       && downsize(&mut sendbuf_size, bytes_to_send).is_ok() {
```

Comment:
Fair enough. I was looking at the call above that has an else if instead of an else, but that is fixed by a later commit. Thanks for taking a look, either solution works for me.